### PR TITLE
Add a test to check that lathe recipes are possible

### PR DESCRIPTION
**Pull request by Tayrtahn**
## About the PR

Adds an integration test that makes sure that lathes are able to accept the materials needed by all of their recipes.

## Why / Balance

To avoid issues like #32206 coming up in the future.

## Technical details

Note that this is going to fail on master currently! As reported in #32206, reinforced glass is listed in the autolathe, but the recipe calls for materials that can't be added to an autolathe. Until that is fixed, this test will fail. Edit: see #33876

The test works by checking each lathe and building a hashset of the materials it accepts (found by checking every material-containing entity against the lathe's storage whitelist). The recipes are then checked to make sure that they don't call for any materials that aren't accepted.

There's also a check to make sure that recipes don't call for a material quantity greater than the capacity of the lathe. No lathes currently have a storage limit, but it's a simple enough check that it doesn't hurt to add it.

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

Not player-facing, so nah.

### CODE REVIEW

File: Content.IntegrationTests/Tests/Lathe/LatheTest.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+                        // Mark the lathe as accepting each material in the entity
+                        foreach (var (material, _) in compositionComponent.MaterialComposition)
+                        {
+                            acceptedMaterials.Add(material);
+                        }
+                    }
+
+                    // Check each recipe assigned to this lathe
+                    foreach (var recipeId in latheComp.StaticRecipes)
```
Comment (slarticodefast) [inline Content.IntegrationTests/Tests/Lathe/LatheTest.cs:77]: The test should also cover the `DynamicRecipes` which are unlocked with research.

**Review by slarticodefast — CHANGES_REQUESTED** [review]
Could you merge master so we can check if the test is successful now that the recipe is fixed?
Also the `AllLatheRecipesValidTest` should probably be moved from `ResearchTest.cs` to the new `Lathetest.cs` for better organisation.

File: Content.IntegrationTests/Tests/Lathe/LatheTest.cs (hunk 1 above)
Comment (slarticodefast) [reply Content.IntegrationTests/Tests/Lathe/LatheTest.cs:77]: Oh, and the additional recipes in the `EmagLatheRecipesComponent` as well.

**Comment by southbridge-fur** [comment]
Frontier has been looking at limiting lathe storage, so the bit that checks against storage limits could be helpful.

**Review by slarticodefast — APPROVED** [review]
Looks good to me.
Very useful PR, thanks!

**Pull request merged**